endaq/batch/analyzer.py

```diff
@@ -73,21 +110,30 @@ def __init__(
             max_key=lambda x: (x.channel.id in preferred_chs, len(x.eventarray)),
         )
 
-        self._filename = dataset.filename
-        self._accelerationFs = None  # gets set in `_accelerationData`
-        self._accel_highpass_cutoff = accel_highpass_cutoff
-        self._accel_start_time = accel_start_time
-        self._accel_end_time = accel_end_time
-        self._accel_start_margin = accel_start_margin
-        self._accel_end_margin = accel_end_margin
-        self._accel_integral_tukey_percent = accel_integral_tukey_percent
-        self._accel_integral_zero = accel_integral_zero
-        self._psd_window = psd_window
-        self._psd_freq_bin_width = psd_freq_bin_width
-        self._pvss_init_freq = pvss_init_freq
-        self._pvss_bins_per_octave = pvss_bins_per_octave
-        self._vc_init_freq = vc_init_freq
-        self._vc_bins_per_octave = vc_bins_per_octave
+        return cls(data, params=params)
+
+    @classmethod
+    def from_raw_data(
+        cls, data: List[Tuple[pd.DataFrame, Tuple[str, str]]], params: CalcParams
+    ):
+        """
+        Instantiate a new `CalcCache` object from raw DataFrame / metadata pairs.
+        """
+
+        @dataclass
+        class DataWrapper:
+            data: pd.DataFrame
+            units: Tuple[str, str]
+
+            def to_pandas(self):
+                self.data.columns.name = "axis"
+                return self.data
```

Comment:
It might be better to define the class outside of the method. In the method, a new class will be generated each time it is run.

Reply:
> The tests could use some descriptive comments

fair point. what tests were you looking at specifically?

Reply:
yeah that personally doesn't bother me so much; it avoids polluting the file namespace with a class that never gets instantiated elsewhere

Reply:
I think I agree with David, this implementation also locks us out of type-checking if that becomes relevant.

Reply:
A possible compromise: the class could be defined within the outer class. It's out of the general package namespace, and can still be explicitly accessed if needed.

Reply:
I don't necessarily agree, but I made the changes anyway. Does this work? 674d91d75bada4ce7a2e6ab4a7b07ef3c2c2445f

Reply:
PR approved -> I assume it's good enough 👍